### PR TITLE
added template code walk-through (WIP)

### PR DESCRIPTION
**Pull request by FranBonath**
Information is to the best of my knowledge, so it might be wrong. Also, there are some files that I don't know what they do, so WIP!

### CODE REVIEW

File: src/content/docs/contributing/nf_core_basic_training.md [hunk 1]
```diff
@@ -76,64 +76,276 @@ To save any file from the explorer panel, right-click the file and select Downlo
...
+
+6. **conf/** and **nextflow.config**
+
+   The `nextflow.config` file is the main config file, it includes all the configurations in `conf/`. Importantly, `conf/` contains the `test.config` file, which is used for pipeline testing. In this course we are not going to touch config files, but they have been extensively covered in the following bytesize talks: [How nf-core configs work (nf-core/bytesize #2)](https://www.youtube.com/watch?v=cXBYusdjrc0&list=PL3xpfTVZLcNiF4hgkW0yXeNzr0d35qlIB&index=8&pp=gAQBiAQB), [Making a new institutional config profile (nf-core/bytesize #10)](https://www.youtube.com/watch?v=Ym1s6sKGzkw&list=PL3xpfTVZLcNiF4hgkW0yXeNzr0d35qlIB&index=9&pp=gAQBiAQB), [nf-core/bytesize: Using nf-core configs in custom pipelines](https://www.youtube.com/watch?v=zgcrI_0SUgg&list=PL3xpfTVZLcNiF4hgkW0yXeNzr0d35qlIB&index=40&pp=gAQBiAQB)
```
Comment (mahesh-panchal) [inline src/content/docs/contributing/nf_core_basic_training.md:292]:
```suggestion
   The `nextflow.config` file is the main config file. In addition to supplying default parameters, it imports all the configurations in `conf/`. Importantly, `conf/` contains the `test.config` file, which is used for pipeline testing. In this course we are not going to touch config files, but they have been extensively covered in the following bytesize talks: [How nf-core configs work (nf-core/bytesize #2)](https://www.youtube.com/watch?v=cXBYusdjrc0&list=PL3xpfTVZLcNiF4hgkW0yXeNzr0d35qlIB&index=8&pp=gAQBiAQB), [Making a new institutional config profile (nf-core/bytesize #10)](https://www.youtube.com/watch?v=Ym1s6sKGzkw&list=PL3xpfTVZLcNiF4hgkW0yXeNzr0d35qlIB&index=9&pp=gAQBiAQB), [nf-core/bytesize: Using nf-core configs in custom pipelines](https://www.youtube.com/watch?v=zgcrI_0SUgg&list=PL3xpfTVZLcNiF4hgkW0yXeNzr0d35qlIB&index=40&pp=gAQBiAQB)
```
Although includeConfig is the keyword in the config, I think imports is more understandable to the reader.

File: src/content/docs/contributing/nf_core_basic_training.md [hunk 2]
```diff
@@ -76,64 +76,276 @@ To save any file from the explorer panel, right-click the file and select Downlo
...
+
+7. **docs/**
+
+   This directory contains additional information to the README file. The most important files are the `output.md` and the `usage.md` files. `usage.md` should descripe what exactly is needed to run the pipeline and `output.md` should descripe all outputs that can be expected. Importantly, for nf-core pipelines, the information from these two files will automatically be displayed on the nf-core website page of the pipeline.
```
Comment (mahesh-panchal) [inline src/content/docs/contributing/nf_core_basic_training.md:296]:
```suggestion
   This directory contains additional information to the README file. The most important files are the `output.md` and the `usage.md` files. `usage.md` should describe what exactly is needed to run the pipeline and `output.md` should describe all outputs that can be expected. Importantly, for nf-core pipelines, the information from these two files will automatically be displayed on the nf-core website page of the pipeline.
```

File: src/content/docs/contributing/nf_core_basic_training.md [hunk 3]
```diff
@@ -76,64 +76,276 @@ To save any file from the explorer panel, right-click the file and select Downlo
...
+
+8. **lib/**
+
+   This directory contains groovy functions and classes that can be imported into the `main.nf` file.
```
Comment (mahesh-panchal) [inline src/content/docs/contributing/nf_core_basic_training.md:300]:
```suggestion
   This directory contains groovy functions and classes that are imported into the `main.nf` file to provide additional functionality not native to Nextflow.
```

File: src/content/docs/contributing/nf_core_basic_training.md [hunk 4]
```diff
@@ -76,64 +76,276 @@ To save any file from the explorer panel, right-click the file and select Downlo
...
+
+9. **modules/local**
+
+   This is were all your custom non-nf-core modules go, we will see an example later in the course
```
Comment (mahesh-panchal) [inline src/content/docs/contributing/nf_core_basic_training.md:304]:
```suggestion
   This is where all your custom non-nf-core modules go. We will cover when and how to make local modules later in the course.
```

File: src/content/docs/contributing/nf_core_basic_training.md [hunk 5]
```diff
@@ -76,64 +76,276 @@ To save any file from the explorer panel, right-click the file and select Downlo
...
+
+10. **modules/nf-core**
+
+    All nf-core modules that are installed using the nf-core tooling will automatically show up in this directory. Keep them here, it is important for automatic updates.
+
+11. **modules.json**
+
```
Comment (mahesh-panchal) [inline src/content/docs/contributing/nf_core_basic_training.md:311]:
```suggestion

This file keeps track of modules installed using nf-core tools from the nf-core/modules repository. This file should only be updated using nf-core tools, and never manually.

```

File: src/content/docs/contributing/nf_core_basic_training.md [hunk 6]
```diff
@@ -76,64 +76,276 @@ To save any file from the explorer panel, right-click the file and select Downlo
...
+    b) _.github/_
+
+    Files in here are used for Continuous integration tests (CI) with github actions as well as other github related defaults, such as a template for issues. We will not touch on these in the course.
+
+    c) _.gitignore_
+
+    d) _.editorconfig_
+
+    e) _.gitpod.yml_
+
```
Comment (mahesh-panchal) [inline src/content/docs/contributing/nf_core_basic_training.md:332]:
```suggestion

This file provides settings to create a Cloud development environment in your browser using Gitpod. It comes installed with the tools necessary to develop and test nf-core pipelines, modules, and subworkflows, allowing you to develop from anywhere without installing anything locally.

```

**Review by mahesh-panchal — COMMENTED** [review]
Some starting comments

**Review by mahesh-panchal — APPROVED** [review]

File: src/content/docs/contributing/nf_core_basic_training.md [hunk 7]
```diff
@@ -76,64 +76,276 @@ To save any file from the explorer panel, right-click the file and select Downlo
...
+                                          `._,._,'
+
+    nf-core/tools version 2.10 - https://nf-co.re
+
+
+? Workflow name demotest
+? Description test pipeline for demo
+? Author FBo
+? Do you want to customize which parts of the template are used? No
+INFO     Creating new nf-core pipeline: 'nf-core/demotest'                                                                                     INFO     Initialising pipeline git repository
```
[the remark this answers is not in the document]
Comment (FranBonath) [reply src/content/docs/contributing/nf_core_basic_training.md:165]: copy paste error, I'll fix it

**Pull request merged**